tests/integration/targets/consul/tasks/main.yml

```diff
@@ -21,9 +21,9 @@
       register: result
       until: result is success
       when: ansible_distribution_file_variety|default() == 'RedHat' and ansible_distribution_major_version is version('6', '<=')
-    - name: Install python-consul
+    - name: Install py-consul
       pip:
-        name: python-consul
+        name: py-consul
         extra_args: "-c {{ remote_constraints }}"
```

Comment:
The failures in CI are due to bugs in older py-consul versions, which didn't correctly specify which Python versions they support. This has been reported in https://github.com/criteo/py-consul/issues/97 and fixed in https://github.com/criteo/py-consul/pull/98, but older versions cannot be fixed, so we have to manually handle this ourselves via constraints.

You need to add corresponding entries to tests/utils/constraints.txt. I think the following should do it (the second is missing some infos):
```
py-consul < 1.3.0 ; python_version < '3.8'  # 1.3.0 dropped support for Python 3.5, 3.6, 3.7
py-consul < 1.5.4 ; python_version < '3.9'  # 1.5.4 dropped support for Python 3.8
```